src/win/fs.c

```diff
@@ -1121,42 +1124,71 @@ void fs__unlink(uv_fs_t* req) {
     }
   }
 
-  if (info.dwFileAttributes & FILE_ATTRIBUTE_READONLY) {
-    /* Remove read-only attribute */
-    FILE_BASIC_INFORMATION basic = { 0 };
-
-    basic.FileAttributes = (info.dwFileAttributes & ~FILE_ATTRIBUTE_READONLY) |
-                           FILE_ATTRIBUTE_ARCHIVE;
-
-    status = pNtSetInformationFile(handle,
-                                   &iosb,
-                                   &basic,
-                                   sizeof basic,
-                                   FileBasicInformation);
-    if (!NT_SUCCESS(status)) {
-      SET_REQ_WIN32_ERROR(req, pRtlNtStatusToDosError(status));
-      CloseHandle(handle);
-      return;
-    }
-  }
+  /* Try posix delete first */
+  disposition_ex.Flags = FILE_DISPOSITION_DELETE | FILE_DISPOSITION_POSIX_SEMANTICS |
+                          FILE_DISPOSITION_IGNORE_READONLY_ATTRIBUTE;
 
-  /* Try to set the delete flag. */
-  disposition.DeleteFile = TRUE;
   status = pNtSetInformationFile(handle,
                                  &iosb,
-                                 &disposition,
-                                 sizeof disposition,
-                                 FileDispositionInformation);
+                                 &disposition_ex,
+                                 sizeof disposition_ex,
+                                 FileDispositionInformationEx);
   if (NT_SUCCESS(status)) {
     SET_REQ_SUCCESS(req);
   } else {
-    SET_REQ_WIN32_ERROR(req, pRtlNtStatusToDosError(status));
+    error = pRtlNtStatusToDosError(status);
+    if (error == ERROR_NOT_SUPPORTED) {
```

Comment:
From https://github.com/microsoft/STL/commit/5735fa1823295da39fc6991c0ca0fdbbe51fa4cb#diff-27a6050161dbeaacef8db1e1a236a18b9762936515f4af930283d8ddb16666baL520-L522 it looks like there are 3 possible failure cases that should be handled here:
```
    case __std_win_error::_Invalid_parameter: // Older Windows versions
    case __std_win_error::_Invalid_function: // Windows 10 1607
    case __std_win_error::_Not_supported: // POSIX delete not supported by the file system
```
```suggestion
    if (error == ERROR_NOT_SUPPORTED /* filesystem does not support posix deletion */ ||
        error == ERROR_INVALID_PARAMETER /* pre Windows 10 error */ ||
        error == ERROR_INVALID_FUNCTION /* pre Windows 10 1607 error */) {
```